tests/test_socket.py

```diff
@@ -111,8 +111,9 @@ def test_socket():
             socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     """)
     result = testdir.runpytest("--verbose", "--disable-socket")
-    result.assert_outcomes(0, 0, 1)
-    assert_socket_blocked(result)
+    result.assert_outcomes(1, 0, 0)
+    with pytest.raises(BaseException):
+        assert_socket_blocked(result)
```

Comment:
It seems there's no need in `assert_socket_blocked` check. With block can be deleted.

Reply:
It's consistent with the other tests this way though.

Reply:
Indeed, but it is strange.  Look at the definition of it
``` python
def assert_socket_blocked(result):
    result.stdout.fnmatch_lines("""
        *SocketBlockedError: A test tried to use socket.socket.*
    """)
```
AFAICT, `assert_socket_blocked()` _does not_ raise an exception ...

https://docs.pytest.org/en/latest/_modules/_pytest/pytester.html#Testdir.runpytest
which can return
https://docs.pytest.org/en/latest/reference.html#_pytest.pytester.RunResult

But when we use fnmatch on the line matcher, I guess that can throw due to pytest fail()?
https://docs.pytest.org/en/latest/reference.html#_pytest.pytester.LineMatcher.fnmatch_lines

Anyway the point is, let's make sure the _logic_ is correct.  The TLDR I think is that `assert_socket_blocked(result)` will throw an exception if socket _IS NOT_ blocked.

So `test_enable_socket_marker` SHOULD NOT BLOCK the socket via `@pytest.mark.enable_socket`, so therefore `assert_socket_blocked(result)` SHOULD be raising an exception.

TBH this is a confusing way to do it, but as @TripleDogDare points out, this is the convention of all other tests.  The "proper" thing to do IMO is write an assertion method called `assert_socket_not_blocked()` (but that is out of scope of this activity).